proposals/pr-contrib-guide.md

```diff
@@ -0,0 +1,490 @@
+# Improvements to the AWS CLI Contribution Process
+
+Proposal         | Metadata
+---------------- | -------------
+**Author**       | Kenneth Daily
+**Status**       | Proposed
+**Created**      | 19-October-2021
+
+## Abstract
+
+This document proposes standards and processes to improve and publish community
+expectations for contributions. It defines the process for proposing,
+submitting, and reviewing a contribution. This proposal will make it easier for
+the community to successfully contribute to the AWS CLI.
+
+## Background and Motivation
+
+### Terminology
+
+In this document, we define a *user* as someone who opens, comments, or
+otherwise interacts with a GitHub issue. We define a *contributor* as someone
+who opens or makes changes to a pull request. We define the *maintainers* as the
+set of individuals who are AWS employees and have the ability to make changes to
+the GitHub repository. We define the *community* as all of the individuals who
+use the AWS CLI.
+
+### Current contribution process
+
+The AWS CLI GitHub repository is available for any individual to report bugs, make
+feature requests, and ask guidance questions through GitHub issues, as well as
+open pull requests to propose code changes. There are currently roughly three
+stages to manage contributions that are common to open source projects: intake,
+implementation, and review.
+
+#### Intake
+
+GitHub issues are triaged regularly to determine that they are correctly
+categorized and express a real and relevant problem or request. An answer is
+provided as soon as possible to acknowledge or resolve the issue. Feature
+requests are reviewed for general suitability and uniqueness, but are not
+actively commented on by maintainers to move the feature forward to a
+contribution from the community. Users can vote for features via "reactions" on
+the issue. The popularity of an issue is occasionally used to determine which
+features the maintainers will work on.
+
+#### Implementation
+
+Users can open pull requests to propose changes that address bugs or feature
+requests. There is no requirement to work on an existing issue in the GitHub
+repository or to discuss a contribution prior to opening a pull request.
+Occassionally guidance is provided to the contributor to improve the proposed
+change.
+#### Review
+
+Pull requests are selected for review opportunistically when the maintainers
+have decided separately that a change or feature should be incorporated.
+Reactions and comments can influence the likelihood of prioritizing a
+contribution, but they do not guarantee it will be reviewed. 
+
+When a feature is selected for review it is added to an internal queue. This
+queue manages the prioritization of these features but is not visible to the
+community. 
+
+The change is automatically tested to indicate if it is compatible with the
+existing code. Maintainers review and request changes to a contributor's GitHub
```

Comment:
This sentence should be moved to the beginning of the "Review" section or possible removed. Where it is right now distracts from the flow of the section. So maybe add it to the beginning of the "Review" section and reword it to something like:

All pull requests are ran through automated tests and checks which provide preliminary feedback on the contribution. Once the contributions passes all automated tests and checks, it requires review from a maintainer to proceed forward toward being merged into the AWS CLI code base.